### ISSUE TITLE
The Opinions

### ISSUE THREAD

**Issue by ahatzz11** [chots/bashy#4]
Since at least a few of us are going to be working on this (and I would love to actually get this thing rocking), I think it's probably important to discuss a little bit about _what_ this should be and _how_ that's done. I see this kind of as an opinionated setup of a shell, specifically centered around `zsh` since [oh-my-zsh](https://github.com/robbyrussell/oh-my-zsh) is so wonderful and because macOS is moving to `zsh` at some point soonish? 

A few things that I think should be figured out as assumptions to use this:
- Entry point
- Folder structure
- Naming

For me I have:
entry point: `~/.zshrc`
zsh: `~/.oh-my-zsh`
custom env stuff and zsh scripts: `~/.omz-custom`

This omz-custom folder is nice because I can make it a git repo (maybe something like this) without it interfering with ALL of zsh.

### .zshrc

```
echo -n "loading zsh..."

export ZSH=/Users/z001l1g/.oh-my-zsh

# Path to custom stuff
ZSH_CUSTOM=~/.omz-custom

# Path to your oh-my-zsh installation.
source $ZSH/oh-my-zsh.sh
# source ~/.target_certs.sh

echo "boom!"

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion

#THIS MUST BE AT THE END OF THE FILE FOR SDKMAN TO WORK!!!
export SDKMAN_DIR="/Users/z001l1g/.sdkman"
[[ -s "/Users/z001l1g/.sdkman/bin/sdkman-init.sh" ]] && source "/Users/z001l1g/.sdkman/bin/sdkman-init.sh"
```

### omz-custom
```
README.md
aliases.zsh
bash-setup.sh
bifrost-vault.zsh
env.zsh
github.zsh
misc.zsh
nyan.zsh
secrets.zsh
sponge.zsh
themes/
zsh-configuration.zsh
zsh_history
```

most of the naming here is for what it is - sponge is @johntosberg's spongebob thing, `github` is all my github functions like `co` and `cm` and `gh`, `alias` has all my aliases, etc.

I also keep my env secrets in a `secrets.zsh` and then add that to the `.gitignore` for protection

What do other people's current environments kind of look like? What should and shouldn't end up being in bashy? is `bashy` even a good name if we're basically all about zsh?

**Comment by DiabetesInc** [comment]
I agree with zsh, also isn't your entrypoint/zsh/custom stuff all just the default way oh-my-zsh does it?
